crawl-ref/source/spl-util.cc

```diff
@@ -1344,8 +1344,12 @@ int spell_highlight_by_utility(spell_type spell, int default_colour,
     {
         return COL_FORBIDDEN;
     }
+    // Grey out spells for which you lack experience or spell levels.
+    if (spell_difficulty(spell) > you.experience_level
+             || player_spell_levels() < spell_levels_required(spell))
+        default_colour = DARKGRAY;
```

Comment:
Thanks for the reminder on unbrace. I hadn't actually set up all the fancy git stuff from docs/develop/git, so it missed it.  fixed now.

and I handled the COL_INAPPLICABLE as well. Looks better now.